source/patterns/@aws-solutions-constructs/core/lib/elasticsearch-helper.ts

```diff
@@ -217,3 +237,122 @@ export function buildElasticSearchCWAlarms(scope: Construct): cloudwatch.Alarm[]
+        endPosition = 1;
+      } else if (twoAvailabilityZoneConfigFlag) {
+        endPosition = 2;
+      } else {
+        endPosition = 3;
+      }
+
+      subnetIds = selectSubnetIds(subnetIds, startPosition, endPosition);
+    } else {
+      endPosition = subnetIds.length;
+
+      if (endPosition > 3) {
+        endPosition = 3;
+      }
+
+      if (endPosition !== 2 && endPosition !== 3) {
+        throw new Error('Error - Availability Zone Count should be set to 2 or 3');
+      }
+
+      subnetIds = selectSubnetIds(subnetIds, startPosition, endPosition);
+    }
+
+    return [{
+      vpcOptions: {
+        subnetIds,
+      },
+    }, subnetIds.length];
+
+  }
+
+  return [{}];
+}
+
+function checkZoneAwareness(props: elasticsearch.CfnDomainProps): boolean | undefined {
+  let zoneAwareness: boolean | undefined = false;
+
+  if (props.elasticsearchClusterConfig && 'zoneAwarenessEnabled' in props.elasticsearchClusterConfig === false &&
+    'zoneAwarenessConfig' in props.elasticsearchClusterConfig) {
+
+    throw Error('Error - zoneAwarenessEnabled must be true to use zoneAwarenessConfig');
+  }
+
+  if (props.elasticsearchClusterConfig) {
+    zoneAwareness = 'zoneAwarenessEnabled' in props.elasticsearchClusterConfig &&
+      props.elasticsearchClusterConfig.zoneAwarenessEnabled === true;
+  }
+
+  return zoneAwareness;
+}
+
+function checkTwoAvailabilityZoneConfig(props: elasticsearch.CfnDomainProps): boolean | undefined {
+  let twoAvailabilityZoneConfig: boolean | undefined = false;
+
+  if (props.elasticsearchClusterConfig) {
+    twoAvailabilityZoneConfig = 'zoneAwarenessConfig' in props.elasticsearchClusterConfig &&
+      props.elasticsearchClusterConfig.zoneAwarenessConfig &&
+      'availabilityZoneCount' in props.elasticsearchClusterConfig.zoneAwarenessConfig &&
+      props.elasticsearchClusterConfig.zoneAwarenessConfig.availabilityZoneCount === 2;
+  }
+
+  return twoAvailabilityZoneConfig;
+}
+
+function selectSubnetIds(subnetIds: string[], startPosition: number, endPosition: number): string[] {
+  return subnetIds.slice(startPosition, endPosition);
+}
+
+function retrievePrivateSubnetIds(vpc: ec2.IVpc) {
+  let targetSubnetType;
+
+  if (vpc.isolatedSubnets.length) {
+    targetSubnetType = ec2.SubnetType.PRIVATE_ISOLATED;
+  } else if (vpc.privateSubnets.length) {
+    targetSubnetType = ec2.SubnetType.PRIVATE_WITH_NAT;
+  } else {
+    throw new Error('Error - ElasticSearch Domains can only be deployed in Isolated or Private subnets');
+  }
+
+  const subnetSelector = {
+    onePerAz: true,
+    subnetType: targetSubnetType
+  };
+
+  return vpc.selectSubnets(subnetSelector).subnetIds;
+}
+
+function buildDefaultZoneAwarenessConfig(subnetIdsLength?: number): elasticsearch.CfnDomainProps {
+  let availabilityZoneCount: number = 3;
+
+  if (subnetIdsLength === 2) {
+    availabilityZoneCount = 2;
+  }
+
+  return {
+    elasticsearchClusterConfig: {
+      zoneAwarenessConfig: {
+        availabilityZoneCount
```

Comment:
Is there a reason not to directly use subnetIdsLength? At this point it is 2 or 3, right?

Reply:
Addressed this in the new code